indra/llcommon/scope_exit.h

```diff
@@ -0,0 +1,34 @@
+/**
+ * @file   scope_exit.h
+ * @author Nat Goodspeed
+ * @date   2024-08-15
+ * @brief  Cheap imitation of std::experimental::scope_exit
+ * 
+ * $LicenseInfo:firstyear=2024&license=viewerlgpl$
+ * Copyright (c) 2024, Linden Research, Inc.
+ * $/LicenseInfo$
+ */
+
+#if ! defined(LL_SCOPE_EXIT_H)
+#define LL_SCOPE_EXIT_H
+
+#include <functional>
+
+namespace LL
+{
+
+class scope_exit
```

Comment:
Did I miss something or `scope_exit` is not used for now?

Reply:
`scope_exit` is not used right now, correct. I introduced it in a previous round of changes to set a Lua recursion flag in `check_interrupts_counter()` and clear it on exit, whether or not it exited via exception.

But since I have often considered the question of whether to introduce a specific class to ensure that some cleanup happens on exit from a block, I added it anyway.